internal/cover/main.go

```diff
@@ -0,0 +1,201 @@
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+// cover is a tool that runs `go test` with cross-package coverage on this
+// repository, ignoring any packages that opt out of coverage with .nocover
+// files. The coverage is written to a coverage.txt file in the current
+// directory.
+//
+// Usage
+//
+// Call cover with a list of one or more import paths of packages being
+// tested.
+//
+//   cover PKG ...
+//
+// This must be run from the root of the project.
+package main
+
+import (
+	"bufio"
+	"errors"
+	"fmt"
+	"go/build"
+	"io"
+	"io/ioutil"
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+)
+
+func main() {
+	log.SetFlags(0)
+	if err := run(); err != nil {
+		log.Fatal(err)
+	}
+}
+
+var (
+	errUsage        = errors.New("usage: cover packages")
+	errNoGoPackage  = errors.New("could not find a Go package in the current directory")
+	errNoImportPath = fmt.Errorf("could not determine import path for the Go package in the current directory")
+)
+
+func run() error {
+	packages := os.Args[1:]
+	if len(packages) == 0 {
+		return errUsage
+	}
+
+	cwd, err := os.Getwd()
+	if err != nil {
+		return fmt.Errorf("could not determine current directory: %v", err)
+	}
+
+	rootPkg, err := build.ImportDir(cwd, 0 /* import mode */)
+	if err != nil {
+		return errNoGoPackage
+	}
+
+	rootImportPath := rootPkg.ImportPath
+	if len(rootImportPath) == 0 {
+		return errNoImportPath
+	}
+
+	// All provided packages must be under rootImport.
+	rootPackagePrefix := rootImportPath + "/"
+	for _, importPath := range packages {
+		if importPath == rootImportPath {
+			continue
+		}
+		if strings.HasPrefix(importPath, rootPackagePrefix) {
+			continue
+		}
+		return fmt.Errorf("%q is not a subpackage of %q", importPath, rootImportPath)
+	}
+
+	covFile, err := ioutil.TempFile("" /* dir */, "coverage")
+	if err != nil {
+		return fmt.Errorf("failed to create temporary file: %v", err)
+	}
+	covFileName := covFile.Name()
+	defer func() {
+		if err := os.Remove(covFileName); err != nil {
```

Comment:
if we fail to remove a temporary file, is it something the user cares about? Is it worth the noise over,
```go
defer os.Remove(covFileName)
```

Reply:
or you could use the `closeFile` function